posthog/dependency_graph.py

```diff
@@ -0,0 +1,535 @@
+import logging
```

Comment:
My main concern is adding dependency resolution logic to the SDKs will make them more complex and harder to maintain. 

Is it possible to move the dependency resolution logic to the server? That way, the local evaluation endpoint returns flags that are already resolved and can be evaluated

Reply:
Oh interesting! If we return the flags in evaluation order, then the SDK logic gets way simpler. Great idea!

We can also transform the filters when sending them to the client so IDs are already resolved to the flag keys.